sdk/core/azure-core-http-netty/pom.xml

```diff
@@ -105,9 +105,12 @@
     </dependency>
 
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <scope>test</scope>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
```

Comment:
Move these two dependencies up into azure-core, and also, make both of these dependencies have test scope.

Reply:
@JonathanGiles sry but what do you mean with 'up into azure-core' the proper parent of all those sdk/core* modules is azure-client-sdk-parent and not the pom.xml located in sdk/core/pom.service.xml. sdk/core/pom.service.xml is an aggregator build which could not be used for dependency inheritance. (afaik)

  `<parent>
    <groupId>com.azure</groupId>
    <artifactId>azure-client-sdk-parent</artifactId>
    <version>1.6.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
    <relativePath>../../../pom.client.xml</relativePath>
  </parent>

`

Reply:
Sorry, I should have been clearer. What I meant to say was that, rather than include junit-jupiter-engine, etc in each module, have it just in the /sdk/core/azure-core/pom.xml file as a dependency, and have all other modules transitively bring that dependency in through that.

Having said that, doing what I suggest here is painful because it means a version bump on azure-core, and all the follow-on from that. I think ignore my comment and leave as-is is best for now, and in the future we can clean this up.

Reply:
But do add test scope :-)

Reply:
hi @JonathanGiles all those new junit5 dependencies are already test-scoped in com.azure:azure-sdk-parent:pom


`<dependency>
        <groupId>org.junit.platform</groupId>
        <artifactId>junit-platform-launcher</artifactId>
        <version>${junit-platform-launcher.version}</version>
        <scope>test</scope>
      </dependency>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-engine</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
      </dependency>
      <dependency>
        <groupId>org.junit.vintage</groupId>
        <artifactId>junit-vintage-engine</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
      </dependency>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-api</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
      </dependency>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-params</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
      </dependency>`

Reply:
This is what is causing the build issues. My recommendation is to remove that test scope from the parent and add it in at the appropriate places.